### PR TITLE
test file pattern

### PR DESCRIPTION
**Pull request by charlierudolph**
@alexdavid @kevgo 

Let mycha accept a test file pattern instead of a test file regex. 
This allows more complex test file patterns. Needed for my current project.

### CODE REVIEW

File: src/test_finder_spec.coffee [hunk 1]
```diff
@@ -30,7 +31,8 @@ describe 'TestFinder', ->
 
     context 'with test file', ->
       beforeEach (done) ->
-        fsExtra.outputFile path.join(@tmpDir, 'server_spec.coffee'), '', (err) =>
+        @filePath = path.join(@tmpDir, 'server_spec.coffee')
```
Comment (kevgo) [inline src/test_finder_spec.coffee]: Remove parentheses

File: src/test_finder.coffee [hunk 2]
```diff
@@ -4,19 +4,22 @@ path = require 'path'
...
+
+    options =
+      cwd: @cwd
+      ignore: '**/node_modules/**'
+
+    callback = (err, files) =>
+      if err then return done err
+      done null, files.map (file) => path.join(@cwd, file)
+
+    glob @testFilePattern, options, callback
```
Comment (kevgo) [inline src/test_finder.coffee]: What is the motivation to separate the callback here? I think it would fit nicely into here if inlined.

**Comment by kevgo** [comment]
Looks good to me!  :ship:

File: src/test_finder.coffee (hunk 2 above)
Comment (charlierudolph) [inline src/test_finder.coffee]: just to make the function call a single line. As long as I'm extracting the options I think it makes sense to do.

Comment (kevgo) [inline src/test_finder.coffee]: What's wrong with this code?

```coffeescript
glob @testFilePattern, options, (err, files) =>
  if err then return done err
  done null, files.map (file) => path.join(@cwd, file)

File: src/test_finder_spec.coffee (hunk 1 above)
Comment (charlierudolph) [inline src/test_finder_spec.coffee]: done

File: src/test_finder.coffee (hunk 2 above)
Comment (charlierudolph) [inline src/test_finder.coffee]: changed

**Pull request closed without merging**